### PR TITLE
Add TX- and SCA-status to psu

### PR DESCRIPTION
**Pull request by isabellaio**
# Summary

Moves the expected TX and SCA status results to the TestPsu data structure. 

Also fixed:
- wrong IBAN (did not belong to any PSU) in the portal sample

# Checklist for Definition of Done

Depending on the task some of the following jobs are not relevant. Please check the items or cross them out (`~~`text`~~`).

* [x] Code is technically reviewed by myself
* [x] Code is manually tested
* [x] Pipeline is green
* [x] Automatic tests for new functionality are created
* [x] ~~Documentation is updated~~
* [x] Branch is rebased and therefore ready to merge
* [x] Git history is clean

### CODE REVIEW

File: service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java [hunk 1]
```diff
@@ -358,15 +369,21 @@ private PsuData initPsuRejected() {
...
         "PSU-Rejected",
         GLOBAL_PASSWORD,
         GLOBAL_TAN,
-        initSingleAccount(accountId, iban, BigDecimal.valueOf(592.59), transaction)
+        initSingleAccount(accountId, iban, BigDecimal.valueOf(592.59), transaction),
+        "Received",
+        "received",
+        "failed",
+        null,
+        "terminatedByTpp",
```
Comment (sw1ld) [inline service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java]: This should also be `null`, shouldn't it?

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java [hunk 2]
```diff
@@ -96,40 +105,47 @@ public void handlePaymentRedirectRequest(String externalId, String psuId,
...
     }
 
-    PisAuthorization payment = pisAuthorization.get();
+    PisAuthorization paymentAuth = pisAuthorization.get();
 
     Optional<List<PisPaymentData>> pisPaymentDataList = pisPaymentDataRepository
-        .findByPaymentId(payment.getPaymentData().getPaymentId());
+        .findByPaymentId(paymentAuth.getPaymentData().getPaymentId());
 
     if (!pisPaymentDataList.isPresent() && pisPaymentDataList.get().isEmpty()) {
```
Comment (sw1ld) [inline service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java]: The two conditions must be connected by `||`, right?

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java [hunk 3]
```diff
@@ -177,4 +193,5 @@ public String getRedirectToTppUriFromAccountRepo(String externalId) {
 
     return tppInfo.getRedirectUri();
   }
+
```
Comment (sw1ld) [inline service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java:196]: remove empty line

File: service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java (hunk 1 above)
Comment (isabellaio) [reply service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java]: done

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java (hunk 2 above)
Comment (isabellaio) [reply service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java]: done

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java (hunk 3 above)
Comment (isabellaio) [reply service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java:196]: done

File: service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java [hunk 4]
```diff
@@ -458,11 +494,79 @@ private PsuData initPsuPending() {
...
+        "finalised",
+        null,
+        null,
+        "finalised"
+    );
+  }
+
+  private TestPsu initPsuConsentExpired() {
+    String iban = "DE86760300801729983660";
+    String accountOwner = "Meryl Streep";
```
Comment (wandi34) [inline service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java:513]: I would use a more neutral and generic name.

File: service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java [hunk 5]
```diff
@@ -458,11 +494,79 @@ private PsuData initPsuPending() {
...
+        "finalised",
+        "Canceled",
+        null,
+        "finalised"
+    );
+  }
+
+  private TestPsu initPsuConsentRevokedByPsu() {
+    String iban = "DE86760300801729983660";
+    String accountOwner = "Tom Cruise";
```
Comment (wandi34) [inline service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java:544]: I would use a more neutral and generic name.

File: service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java [hunk 6]
```diff
@@ -458,11 +494,79 @@ private PsuData initPsuPending() {
...
+    Transaction transaction = new Transaction(
+        "3bf5b19b-12e9-4ab4-bdac-a5c7695ae4b9",
+        BigDecimal.valueOf(199.99),
+        EUR,
+        LocalDate.parse("2018-09-20"),
+        accountOwner,
+        iban,
+        "Amazon",
+        "",
+        "Amazon.de: Ihre Bestellung #4528499"
```
Comment (wandi34) [inline service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java:556]: Maybe use other descriptions. The project is open source, i would use more neutral data.

File: service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java (hunk 4 above)
Comment (sw1ld) [reply service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java:513]: We will change the names and transactions in a following separate commit.

File: service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java (hunk 6 above)
Comment (sw1ld) [reply service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java:556]: We will change the names and transactions in a following separate commit.

File: service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java (hunk 5 above)
Comment (sw1ld) [reply service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java:544]: We will change the names and transactions in a following separate commit.

**Pull request merged**